octokit/repositories.go

```diff
@@ -1,57 +1,91 @@
 package octokit
 
 import (
-	"net/url"
 	"time"
 
 	"github.com/jingweno/go-sawyer/hypermedia"
 )
 
 // Hyperlinks to the various repository locations on github.
-// RepositoryURL is a template for a particular repository for a particular owner.
-// ForksURL is a template for the forks of a user's repository.
-// UserRepositoriesURL is the address for all user repositories.
-// OrgRepositoriesUrl is the template for repositories within a particular organization.
 var (
-	RepositoryURL       = Hyperlink("repos/{owner}/{repo}")
-	ForksURL            = Hyperlink("repos/{owner}/{repo}/forks")
+	// RepositoryURL is a template for a particular repository for a
+	// particular owner.
+	RepositoryURL = Hyperlink("repos/{owner}/{repo}")
+
+	// ForksURL is a template for the forks of a user's repository.
+	ForksURL = Hyperlink("repos/{owner}/{repo}/forks")
+
+	// UserRepositoriesURL is the address for all user repositories.
 	UserRepositoriesURL = Hyperlink("user/repos")
-	OrgRepositoriesURL  = Hyperlink("orgs/{org}/repos")
+
+	// OrgRepositoriesUrl is the template for repositories within a
+	// particular organization.
+	OrgRepositoriesURL = Hyperlink("orgs/{org}/repos")
+
+	// CollaboratorsURL is the template for accessing the collaborators
+	// of a particular repository.
+	CollaboratorsURL = Hyperlink(
+		"repos/{owner}/{repo}/collaborators/{username}")
 )
 
 // Repositories creates a RepositoriesService with a base url
-func (c *Client) Repositories(url *url.URL) (repos *RepositoriesService) {
-	repos = &RepositoriesService{client: c, URL: url}
+func (c *Client) Repositories() (repos *RepositoriesService) {
+	repos = &RepositoriesService{client: c}
 	return
 }
 
-// RepositoriesService is a service providing access to repositories from a particular url
+// RepositoriesService is a service providing access to repositories from a
+// particular url
 type RepositoriesService struct {
 	client *Client
-	URL    *url.URL
 }
 
 // One gets a specific repository based on the url of the service
-func (r *RepositoriesService) One() (repo *Repository, result *Result) {
-	result = r.client.get(r.URL, &repo)
+func (r *RepositoriesService) One(uri *Hyperlink, params M) (repo *Repository,
+	result *Result) {
+	if uri == nil {
+		uri = &RepositoryURL
+	}
+	url, err := uri.Expand(params)
+	if err != nil {
+		return nil, &Result{Err: err}
+	}
+	result = r.client.get(url, &repo)
 	return
 }
 
 // All gets a list of all repositories associated with the url of the service
-func (r *RepositoriesService) All() (repos []Repository, result *Result) {
-	result = r.client.get(r.URL, &repos)
+func (r *RepositoriesService) All(uri *Hyperlink, params M) (repos []Repository,
+	result *Result) {
+	if uri == nil {
+		uri = &RepositoryURL
```

Comment:
Shouldn't this default to a listing resource URL like `/repositories` (list all public repos) or `/user/repos` (list mine)?